### ISSUE TITLE
gopass-generated config `path` does not accept a relative path

### ISSUE THREAD

**Issue by siriobalmelli** [gopasspw/gopass#1624]
### Summary

A config `path` entry of `~/.password-store` will not parse, whereas an absolute path eg `/Users/sirio/.password-store` will work.

### Steps To Reproduce

1. Remove existing config:

        rm ~/.config/gopass/config.yml

2. Use `gopass` to generate a config:

        $ gopass config path "~/.password-store"
        path: ~/.password-store

    This will generate the following config:

        autoclip: false
        autoimport: true
        cliptimeout: 45
        exportkeys: true
        mime: true
        nocolor: false
        nopager: false
        notifications: true
        path: ~/.password-store
        safecontent: false
        mounts: {}

3. Generated config fails to parse

        $ gopass list
        It seems you are new to gopass. Do you want to run the onboarding wizard? [Y/n/q]:

4. Change relative path to absolute and it will now parse

        $ sed -i"" "s|~|$HOME|g" ~/.config/gopass/config.yml
        $ gopass list
        # ... list output ...

### Expected behavior

Relative paths in config should be correctly parsed.

It is important to support relative paths:

- user home directories don't have the same path between Darwin/Linux or between same-OS machines where the username is different.
- it should be possible to maintain `~/.config` as a Git repo sync'ed between multiple machines

### Environment

- OS: Darwin (macOS Catalina)

- OS version: `Darwin computer.home 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64 i386 MacBookPro16,1 Darwin`

- gopass Version:
```
gopass 1.10.1+fc5e9b75254fd975ff458cf4843f2def21f872ee (fc5e9b75254fd975ff458cf4843f2def21f872ee) go1.15.2 darwin amd64
<root>     - gpg 2.2.23 - git 2.28.0
Available Crypto Backends: age, gpgcli, plain
Available Storage Backends: fs, gitfs, ondisk
```

- Installation method: Nixpkgs (building latest commit from source)

**Comment by siriobalmelli** [comment]
This appears related to https://github.com/gopasspw/gopass/issues/872

**Comment by siriobalmelli** [comment]
This is still existing with the latest commit:

```bash
$ gopass --version
gopass 1.10.1+af9c95d20b722dee331adcc3cc1ab1ce0108dc2b (af9c95d20b722dee331adcc3cc1ab1ce0108dc2b) go1.15.6 darwin amd64
```

**Comment by dominikschulz** [comment]
The config is parsed fine. And then gopass tries to use the literal value "~" as a directory, creating it below the current directory.

This needs to be fixed for the new patch release.
We need to make sure to expand the path before using it, but not writing the expanded value back to the config.

**Comment by dominikschulz** [comment]
This should be fixed by #1713. The onboarding wizard will still write an absolute path to the config, but you can easily turn that into a relative one and gopass shouldn't overwrite that later.
